contracts/plugins/assets/RTokenAsset.sol

```diff
@@ -128,8 +139,46 @@ contract RTokenAsset is IAsset, VersionedAsset {
 
     // solhint-enable no-empty-blocks
 
+    function forceUpdatePrice() external {
+        _updateCachedPrice();
+    }
+
+    function latestPrice() external returns (uint192 rTokenPrice, uint256 updatedAt) {
+        // Situations that require an update, from most common to least common.
+        if (
+            cachedOracleData.cachedAtTime + ORACLE_TIMEOUT <= block.timestamp || // Cache Timeout
+            cachedOracleData.cachedAtNonce != basketHandler.nonce() || // Basket nonce was updated
+            cachedOracleData.cachedTradesNonce != backingManager.tradesNonce() || // New trades were started
+            cachedOracleData.cachedTradesOpen != backingManager.tradesOpen() // ..or settled
+        ) {
+            _updateCachedPrice();
+        }
+
+        return (cachedOracleData.cachedPrice, cachedOracleData.cachedAtTime);
+    }
+
     // ==== Private ====
 
+    // Update Oracle Data
+    function _updateCachedPrice() internal {
+        if (cachedOracleData.cachedAtTime == block.timestamp) {
```

Comment:
I don't think we want to skip here. Consider the case where earlier in the block there was a `latestPrice()` call followed by a `rebalance()` / `settleTrade()`. We'd enter this internal function and then skip updating the price, even though token balances did in fact move over a boundary since the price was last cached.